### PR TITLE
New dribbler skill

### PR DESCRIPTION
**Pull request by timovdk**
Added a new dribbler skill to prepare to move to the STP model

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -163,8 +163,10 @@ set(SKILLS_SOURCES
         ${PROJECT_SOURCE_DIR}/src/skills/Wait.cpp
         ${PROJECT_SOURCE_DIR}/src/skills/KickTo.cpp
         ${PROJECT_SOURCE_DIR}/src/skills/formations/StopFormation.cpp
-        ${PROJECT_SOURCE_DIR}/src/skills/formations/BallPlacementFormation.cpp
-        )
+        ${PROJECT_SOURCE_DIR}/src/skills/formations/BallPlacementFormation.cpp)
+
+set(NEW_SKILLS_SOURCES
```
Comment (Yuhanun) [inline CMakeLists.txt:169]: NEW_SKILL_SOURCES -> SKILL_SOURCES (unless it already exists

File: CMakeLists.txt [hunk 2]
```diff
@@ -163,8 +163,10 @@ set(SKILLS_SOURCES
...
+        ${PROJECT_SOURCE_DIR}/src/new_skills/SetDribbler.cpp)
```
Comment (Yuhanun) [inline CMakeLists.txt]: ) on the next line

File: include/roboteam_ai/new_skills/SetDribbler.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+#define RTT_SETDRIBBLER_H
+
+#include <include/roboteam_ai/skills/Skill.h>
+
+namespace rtt::ai {
+
+/**
+ * This skill sets the dribbler at a speed specified in the blackboard
+ */
+
```
Comment (Yuhanun) [inline include/roboteam_ai/new_skills/SetDribbler.h]: Unnessecary blank line

File: include/roboteam_ai/new_skills/SetDribbler.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+class SetDribbler : public Skill {
+   public:
+    explicit SetDribbler(std::string name, bt::Blackboard::Ptr blackboard);
+    void onInitialize() override;
+    Status onUpdate() override;
+    void onTerminate(Status s) override;
```
Comment (Yuhanun) [inline include/roboteam_ai/new_skills/SetDribbler.h]: Document these, which property of the blackboard does it use etc...

Comment (Yuhanun) [reply include/roboteam_ai/new_skills/SetDribbler.h]: Take a look at Doxygen

File: src/new_skills/SetDribbler.cpp [hunk 5]
```diff
@@ -0,0 +1,23 @@
+//
+// Created by timovdk on 2/11/20.
+//
+
+#include "new_skills/SetDribbler.h"
+
+namespace rtt::ai {
+
+SetDribbler::SetDribbler(std::string name, bt::Blackboard::Ptr blackboard) : Skill(std::move(name), std::move(blackboard)) {}
```
Comment (Yuhanun) [inline src/new_skills/SetDribbler.cpp]: Apply clang-format

**Review by Yuhanun — APPROVED** [review]
Looks fine :)

File: include/roboteam_ai/new_skills/SetDribbler.h [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+#include <include/roboteam_ai/skills/Skill.h>
+
+namespace rtt::ai {
+
+/**
+ * This skill sets the dribbler at a speed specified in the blackboard
+ */
+
+class SetDribbler : public Skill {
```
Comment (rolfvdhulst) [inline include/roboteam_ai/new_skills/SetDribbler.h:15]: I highly suggest if we want to make new skills to also make a new base class or at least refactor it.  Your code is mostly fine, but this base class is really ugly.

Comment (Yuhanun) [reply include/roboteam_ai/new_skills/SetDribbler.h:15]: I haven't seen it yet, ill check that out in a sec

Comment (timovdk) [reply include/roboteam_ai/new_skills/SetDribbler.h:15]: I will ask Jesse tomorrow, I was thinking the same

File: CMakeLists.txt (hunk 1 above)
Comment (timovdk) [reply CMakeLists.txt:169]: SKILL_SOURCES already exist, for clarity the NEW_SKILL_SOURCES will be used until the old skills are deprecated

**Comment by emielsteerneman** [comment]
I'm ready to merge this but I'm confused about how this works. It sets the dribbler to a certain speed, and then publishes the command. What about other parts of the command such as id, velocity, rotation?

**Pull request merged**